contentcuration/contentcuration/views.py

```diff
@@ -157,6 +161,73 @@ def file_create(request):
             "object_id": new_node.pk
         }))
 
+# TODO-BLOCKER: remove this csrf_exempt! People might upload random stuff here and we don't want that.
+@csrf_exempt
+def file_diff(request):
+    logging.debug("Entering the file_diff endpoint")
+    if request.method != 'POST':
+        raise HttpResponseBadRequest("Only POST requests are allowed on this endpoint.")
+    else:
+        data = json.loads(request.body)
+        return HttpResponse(json.dumps(get_file_diff(data)))
+
+# TODO-BLOCKER: remove this csrf_exempt! People might upload random stuff here and we don't want that.
+@csrf_exempt
+def api_file_upload(request):
+    if request.method != 'POST':
+        raise HttpResponseBadRequest("Only POST requests are allowed on this endpoint.")
+    else:
+        try:
+            fobj = request.FILES["file"]
+            file_path = generate_file_on_disk_name(fobj._name.split(".")[-2], fobj._name)
+
+            with open(file_path, 'wb') as destf:
+                shutil.copyfileobj(fobj, destf)
+
+            return HttpResponse(json.dumps({
+                "success": True,
+            }))
+        except KeyError:
+            raise ObjectDoesNotExist("Missing attribute from data")
+
+# TODO-BLOCKER: remove this csrf_exempt! People might upload random stuff here and we don't want that.
+@csrf_exempt
+def api_create_channel_endpoint(request):
+    if request.method != 'POST':
+        raise HttpResponseBadRequest("Only POST requests are allowed on this endpoint.")
+    else:
+        data = json.loads(request.body)
+        try:
+            content_data = data['content_data']
+            channel_data = data['channel_data']
+            file_data = data['file_data']
+
+            obj = api_create_channel(channel_data, content_data, file_data)
+            invitation = Invitation.objects.create(channel=obj)
+
+            return HttpResponse(json.dumps({
+                "success": True,
+                "new_channel": obj.pk,
+                "invite_id":invitation.pk,
```

Comment:
The invitations are used for handling inviting other editors to a channel. I tried adding a pending editors many-to-many field, but there was the case where a user could invite someone who hadn't created an account yet. The invitation id acts as a unique key that would make it difficult for anyone to be able to access the channel without being invited. Not sure if there is a way to use OAuth tokens, but I can add them when we start looking at permissions